robotbase/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java

```diff
@@ -0,0 +1,383 @@
+   *                                  CAN FD, and 100 Hz on CAN 2.0.
+   * @param odometryStandardDeviation The standard deviation for odometry
+   *                                  calculation
+   *                                  in the form [x, y, theta]ᵀ, with units in
+   *                                  meters
+   *                                  and radians
+   * @param visionStandardDeviation   The standard deviation for vision
+   *                                  calculation
+   *                                  in the form [x, y, theta]ᵀ, with units in
+   *                                  meters
+   *                                  and radians
+   * @param modules                   Constants for each specific module
+   */
+  public CommandSwerveDrivetrain(
+    SwerveDrivetrainConstants drivetrainConstants,
+    double odometryUpdateFrequency,
+    Matrix<N3, N1> odometryStandardDeviation,
+    Matrix<N3, N1> visionStandardDeviation,
+    SwerveModuleConstants<?, ?, ?>... modules
+  ) {
+    super(
+      drivetrainConstants,
+      odometryUpdateFrequency,
+      odometryStandardDeviation,
+      visionStandardDeviation,
+      modules
+    );
+    if (Utils.isSimulation()) {
+      startSimThread();
+    }
+  }
+
+  /**
+   * Returns a command that applies the specified control request to this swerve
+   * drivetrain.
+   *
+   * @param request Function returning the request to apply
+   * @return Command to run
+   */
+  public Command applyRequest(Supplier<SwerveRequest> requestSupplier) {
+    return run(() -> this.setControl(requestSupplier.get()));
+  }
+
+  /**
+   * Runs the SysId Quasistatic test in the given direction for the routine
+   * specified by {@link #m_sysIdRoutineToApply}.
+   *
+   * @param direction Direction of the SysId Quasistatic test
+   * @return Command to run
+   */
+  public Command sysIdQuasistatic(SysIdRoutine.Direction direction) {
+    return m_sysIdRoutineToApply.quasistatic(direction);
+  }
+
+  /**
+   * Runs the SysId Dynamic test in the given direction for the routine
+   * specified by {@link #m_sysIdRoutineToApply}.
+   *
+   * @param direction Direction of the SysId Dynamic test
+   * @return Command to run
+   */
+  public Command sysIdDynamic(SysIdRoutine.Direction direction) {
+    return m_sysIdRoutineToApply.dynamic(direction);
+  }
+
+  @Override
+  public void periodic() {
+    /*
+     * Periodically try to apply the operator perspective.
+     * If we haven't applied the operator perspective before, then we should apply
+     * it regardless of DS state.
+     * This allows us to correct the perspective in case the robot code restarts
+     * mid-match.
+     * Otherwise, only check and apply the operator perspective if the DS is
+     * disabled.
+     * This ensures driving behavior doesn't change until an explicit disable event
+     * occurs during testing.
+     */
+    if (!m_hasAppliedOperatorPerspective || DriverStation.isDisabled()) {
```

Comment:
A future enhancement would be to remove this from the periodic and move it somewhere else (a command), and then also add buttons to the driver controller to reset and flip operator perspective.